### PR TITLE
feat: remove optional fields from register form

### PR DESCRIPTION
**Pull request by zainab-amir**
This is part of the larger ticket where we will move optional fields to progressive profiling page.

[VAN-837](https://openedx.atlassian.net/browse/VAN-837)

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -23,7 +23,6 @@ AUTHN_MINIMAL_HEADER=true
 LOGIN_ISSUE_SUPPORT_LINK='/login-issue-support-url'
 TOS_AND_HONOR_CODE='http://localhost:18000/honor'
 PRIVACY_POLICY='http://localhost:18000/privacy'
-REGISTRATION_OPTIONAL_FIELDS=''
```
Comment (waheedahmed) [inline .env.development:26]: Are we defining a new variable for optional fields on the progressive profiling page?

File: src/register/OptionalFields.jsx [hunk 2]
```diff
@@ -1,110 +0,0 @@
...
-
-import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
-import { getConfig } from '@edx/frontend-platform';
-import { Form, Icon } from '@edx/paragon';
-import { ExpandMore } from '@edx/paragon/icons';
-
-import { EDUCATION_LEVELS, GENDER_OPTIONS, YEAR_OF_BIRTH_OPTIONS } from './data/constants';
-import messages from './messages';
-
-const OptionalFields = (props) => {
```
Comment (waheedahmed) [inline src/register/OptionalFields.jsx:12]: Instead of removing, can we utilize these on the progressive profiling page in some way?

**Review by waheedahmed — APPROVED** [review]
Couple of questions, otherwise 👍

File: .env.development (hunk 1 above)
Comment (zainab-amir) [reply .env.development:26]: none yet. We might not need it.

File: src/register/OptionalFields.jsx (hunk 2 above)
Comment (zainab-amir) [reply src/register/OptionalFields.jsx:12]: We will probably not need this code for our approach but the reason for creating this PR was to remove the fields from Registration folder. These are already duplicated in progressive profiling folder.

**Pull request merged**